src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs

```diff
@@ -499,6 +499,53 @@ internal void Insert(ref T[] list, string? name, MemberListType listType)
                     }
                 }
 
+                private void MergeWithGlobalListInOrder(T[] list)
+                {
+                    T[]? cachedMembers = m_allMembers;
+
+                    if (cachedMembers == null)
+                    {
+                        m_allMembers = list;
+                        return;
+                    }
+
+                    int length = list.Length;
+                    T[] newCache = list;
+
+                    // When all members are requested there should be no element in cache that not found within all members list.
+                    // But somehow in very rare case such element found in cache and not appending that to the all members list
+                    // causing random test failure, must be related to the bug commented on row 597
+                    Array.Resize(ref newCache, length + 1);
```

Comment:
Duplicating my feedback in the original branch: this `+1` and the `!foundInList` below seem to indicate that one 1 missing member is supported.

Reply:
Yes, technically there should be no any, as mentioned in the comment somehow there is extra element causing a bug mentioned here: https://github.com/dotnet/runtime/blob/5891a41d36b1e05c0fed78e4648105cac8a81776/src/coreclr/System.Private.CoreLib/src/System/RuntimeType.CoreCLR.cs#L591-L602 
This is the copy of this workaround, but it might not relevant for `MemberListType.All`, then I will remove this

Reply:
Do we know how to add these "extra" elements? Such as:
- Known issues with `MethodInfo.Equals()` allowing more than one `MethodInfo` instance for a given  method depending on how it was obtained (I'm not sure if both would get added to this cache however).
- Concurrency issues where 2 MethodInfo instances could be created in a race condition for a given method?

Reply:
> Do we know how to add these "extra" elements? Such as:
> - Known issues with MethodInfo.Equals() allowing more than one MethodInfo instance for a given method depending on how it was obtained (I'm not sure if both would get added to this cache however).

Both would added to the cache, I think in this should return same instance no matter how it was created (if it doable of course)

> - Concurrency issues where 2 (or more) MethodInfo instances could be created in a race condition for a given method?

I suspect there is concurrency issue because of random/inconsistent failures in some projects, but not sure where/how it is happening

Reply:
> concurrency issue

There is a big lock around this whole method. The logic in this method is expected to discard the second MethodInfo instance. We should have just the one winning MethodInfo instance when this method exits.

Reply:
Removed this extra handling